pkg/testutils/jobutils/backup_restore.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobutils
+
+// getExternalBytesQuery returns the count of external bytes over the tenant that ran the query.
+const GetExternalBytesQuery = `SELECT COALESCE(stats->>'external_file_bytes','0') FROM crdb_internal.tenant_span_stats(
+	ARRAY(SELECT(crdb_internal.tenant_span()[1], crdb_internal.tenant_span()[2])))`
```

Comment:
I'm not sure this is the right span for `restore tenant` verification, specifically I think the end key should maybe just be \xff\xff

Reply:
i'm not sure it matters? i could just bors this and move on with life ?

Reply:
like, if this is run by the newly restored app tenant, then this is checking the right key space i think

Reply:
but if you run this in the system tenant -- i.e. if your test issues `restore` on a tc.Conns[0] and then turns around and runs this helper on that same conn, they'll get a spurious 0 even while the restore job is still downloading since the helper is looking only at the span of the system tenant. In a way making it a helper has made this worse than it was before, since you won't see the query at the callsite and realize your mistake.

Reply:
if you just replace `tenant_span[1]` and `tenant_span[2]` with \x89 and \xffff then it is reliably checking the whole cluster -- all tables in all tenants -- and the potential for mistakes is gone, right ?

Reply:
the query `SELECT COALESCE(stats->>'external_file_bytes','0') FROM crdb_internal.tenant_span_stats(
		ARRAY(SELECT('\x89'::BYTES,'\xffff'::BYTES)))`

fails when run from the system tenant with ` pq: rpc error: code = Unauthenticated desc = requested key span /{Table/1-Max} not fully contained in tenant keyspace /Tenant/2{0-1}`

I've renamed the const to `getExternalBytesForConnectedTenant` to prevent test writers from falling for the trap you laid out.

I feel like this is a pretty small limitation.  If you feel strongly that this isn't sufficient, can i leave a todo for you to address? I feel that i've worked on this much longer than i planned to and would like to work on to more important things.

Reply:
the query `SELECT COALESCE(stats->>'external_file_bytes','0') FROM crdb_internal.tenant_span_stats(
ARRAY(SELECT('\x89'::BYTES,'\xffff'::BYTES)))` can only get run from the system tenant. If we only provided that query, then metamorphic tests which run online restore from an app tenant could not run this check. 
		
Check out the latest version of the patch, where i provide both tenant_span() query and the tenanttablemin/max query.